sdk/core/azure-core-amqp/samples/eventhub_token_writer_sample/eventhub_token_writer_sample.cpp

```diff
@@ -12,45 +12,57 @@
 #include <limits>
 #include <string>
 
-#define EH_HOST "<<<Replace with the eventhubs host name>>>"
-#define EH_ENTITY "<<<Replace with the eventhub name>>>"
-
 #define EH_AUTHENTICATION_SCOPE "https://eventhubs.azure.net/.default"
 
-#define EH_ENTITY_URL "amqps://" EH_HOST "/" EH_ENTITY
-
 int main()
 {
+  // Retrieve the eventhub connection string so we can extract the host name and entity name. We are
+  // NOT using the connection string to connect to the eventhub.
+  std::string eventhubConnectionString
+      = Azure::Core::_internal::Environment::GetVariable("EVENTHUB_CONNECTION_STRING");
+  Azure::Core::Amqp::_internal::ConnectionStringParser connectionStringCredential(
+      eventhubConnectionString);
+  std::string eventhubsHost = connectionStringCredential.GetHostName();
+  std::string eventhubsEntity = connectionStringCredential.GetEntityPath();
+
+  // If the connection string does not specify an entity path, then look for the eventhub name in an
+  // environment variable.
+  if (eventhubsEntity.empty())
+  {
+    eventhubsEntity = Azure::Core::_internal::Environment::GetVariable("EVENTHUB_NAME");
+  }
+
+  // Establish credentials for the eventhub client.
+  auto credential{std::make_shared<Azure::Identity::ClientSecretCredential>(
```

Comment:
I'm using ClientSecretCredential because `EnvironmentCredential` doesn't work with the new-TestResources.ps1 script, and neither does DefaultAzureCredential.

Reply:
Why can't the resources deployment script to deploy variables as `AZURE_xxxx` instead of `SAMPLE_xxx`? Then, all the credentials will work, and the sample will be about using AMQP, and so much attention won;t be focused on using the ClientSecretCredential class specifically.

Reply:
That is a question that you need to bring up with the engineering systems team.  The new-testresources.ps1 script creates environment variables named '<SERVICE>_XXX'. For the samples, if you use the invocation specified in the readme.md file, then it creates a "service" named SAMPLES, thus SAMPLE_CLIENT_ID, etc.